eclair-core/src/main/scala/fr/acinq/eclair/api/EclairClient.scala

```diff
@@ -0,0 +1,227 @@
+
+
+  implicit val dispatcher = system.dispatcher
+
+  private val _ = require(datadir.mkdirs(), s"Failed to make directory for eclair ${datadir.getAbsolutePath}")
+
+  private val setup = new Setup(datadir, actorSystem = system)
+  private val kitF: Future[Kit] = setup.bootstrap
+  private val logger = LoggerFactory.getLogger(this.getClass.getSimpleName)
+
+  def nodeParams: NodeParams = setup.nodeParams
+
+  override def getBalance: Future[Satoshi] = kitF.flatMap(_.wallet.getBalance)
+
+  override def createInvoice(amt: MilliSatoshi, description: String = ""): Future[PaymentRequest] = {
+    val recv = ReceivePayment(Some(MilliSatoshi(amt.toLong)), description)
+    receive(recv)
+  }
+
+
+  override def payInvoice(invoice: PaymentRequest): Future[PaymentResult] = {
+    kitF.flatMap { kit =>
+      val sendPayment = SendPayment(invoice.amount.get.toLong,invoice.paymentHash,invoice.nodeId)
+      kit.paymentInitiator.ask(sendPayment).mapTo[PaymentResult]
+    }
+  }
+
+  override def checkPayment(invoice: PaymentRequest): Future[Boolean] = {
+    checkPayment(invoice.paymentHash)
+  }
+
+  override def connectToNode(nodeURI: NodeURI): Future[String] = {
+    logger.debug(s"${setup.nodeParams.nodeId} is connecting to ${nodeURI.nodeId}")
+    kitF.flatMap { kit =>
+      val conn = (kit.switchboard.ask(Peer.Connect(nodeURI))).mapTo[String]
+      conn
+    }
+  }
+
+
+  override def getNodeURI: NodeURI = {
+    val address = setup.nodeParams.publicAddresses.head
+    val hostAndPort =  HostAndPort.fromParts(address.getHostString,address.getPort)
+    new NodeURI(setup.nodeParams.nodeId,hostAndPort)
+  }
+
+  override def openChannel(openChannelMsg: Peer.OpenChannel): Future[String] = {
+    logger.debug(s"${setup.nodeParams.nodeId} is opening a channel with ${openChannelMsg.remoteNodeId}")
+    kitF.flatMap { kit =>
+      val channel = kit.switchboard.ask(openChannelMsg).mapTo[String]
+      channel
+    }
+  }
+
+  override def getPeers: Future[Vector[Peer.PeerInfo]] = {
+    kitF.flatMap { kit =>
+      val result: Future[Iterable[PeerInfo]] = for {
+        peers <- (kit.switchboard ? 'peers).mapTo[Map[PublicKey, ActorRef]]
+        peerinfos <- Future.sequence(peers.values.map(peer => (peer ? GetPeerInfo).mapTo[PeerInfo]))
+      } yield peerinfos
+      result.map(_.toVector)
+    }
+  }
+
+  override def close(id: String, spk: Option[BinaryData]): Future[String] = {
+    sendToChannel(id, CMD_CLOSE(scriptPubKey = spk)).mapTo[String]
+  }
+
+  override def forceClose(id: String): Future[String] = {
+    sendToChannel(id, CMD_FORCECLOSE).mapTo[String]
+  }
+
+  override def channels(pubKeyOpt: Option[PublicKey]): Future[Vector[LocalChannelInfo]] = {
+    kitF.flatMap { kit =>
+
+      val channelsIdF: Future[Iterable[BinaryData]] = (kit.register ? 'channels)
+        .mapTo[Map[BinaryData, ActorRef]]
+        .map { m :Map[BinaryData, ActorRef] =>
+          if (pubKeyOpt.isDefined) {
```

Comment:
Review needed here, I think this is correct but would like more eye balls on it.